contracts/modules/token/TokenFunctions.sol

```diff
@@ -112,26 +102,33 @@ contract TokenFunctions is Iupgradable {
    * @return the status of the successful burning
    */
   function burnDepositCN(uint coverId) public onlyInternal returns (bool success) {
+
     address _of = qd.getCoverMemberAddress(coverId);
-    uint amount;
-    (amount,) = td.depositedCN(coverId);
-    amount = (amount.mul(50)).div(100);
     bytes32 reason = keccak256(abi.encodePacked("CN", _of, coverId));
-    tc.burnLockedTokens(_of, reason, amount);
-    success = true;
+    uint lockedAmount = tc.tokensLocked(_of, reason);
+
+    (uint amount,) = td.depositedCN(coverId);
+    amount = amount.div(2);
+
+    // limit burn amount to actual amount locked
+    uint burnAmount = lockedAmount < amount ? lockedAmount : amount;
```

Comment:
i asked this before i think, is there any problem if your deposit CN is not an even number and there's 1 wei here left after burning twice (!= 0)

Reply:
In the old setup I came to the conclusion that it was fine. Worth taking another look at it with the new code.